products/zoningtaxlots/models/intermediate/int__zoningtaxlots.sql

```diff
@@ -0,0 +1,556 @@
+{{ config(
+    materialized = 'table'
+) }}
+
+WITH dof_dtm AS (
+    SELECT * FROM {{ ref('stg__dof_dtm') }}
+),
+
+commercialoverlay AS (
+    SELECT * FROM {{ ref('int__commercialoverlay') }}
+),
+
+specialpurpose AS (
+    SELECT * FROM {{ ref('int__specialpurpose') }}
+),
+
+limitedheight AS (
+    SELECT * FROM {{ ref('int__limitedheight') }}
+),
+
+zoningmapindex AS (
+    SELECT * FROM {{ ref('int__zoningmapindex') }}
+),
+
+zoningdistricts AS (
+    SELECT * FROM {{ ref('int__zoningdistricts') }}
+),
+
+inwoodrezooning AS (
+    SELECT * FROM {{ ref('int__inwoodrezooning') }}
+),
+
+zonechange AS (
+    SELECT * FROM {{ ref('int__inzonechange') }}
+),
+
+-- insert bbl
+
+insertion AS (
+    SELECT
+        id AS dtm_id,
+        bbl,
+        boro AS boroughcode,
+        block AS taxblock,
+        lot AS taxlot,
+        ST_AREA(geom) AS area
+    FROM dof_dtm
+),
+
+-- populate bbl field if it's NULL
+
+pop_bbl AS (
+    SELECT
+        dtm_id,
+        (CASE
+            WHEN bbl IS NULL OR LENGTH(bbl) < 10
+                THEN boroughcode || LPAD(taxblock, 5, '0') || LPAD(taxlot, 4, '0')::text
+            ELSE bbl
+        END) AS bbl,
+        boroughcode,
+        taxblock,
+        taxlot,
+        area
+    FROM insertion
+),
+
+-- add commercialoverlay
+
+add_comm1 AS (
```

Comment:
One final request - can we group these by category? So for the "add_comm" queries, instead of one CTE per row number, do something with
```sql
FROM pop_bbl AS ztl
LEFT JOIN commercialoverlay AS co1
    ON ztl.dtm_id = co.dtm_id AND co.row_number = 1
LEFT JOIN commercialoverlay AS co2
    ON ztl.dtm_id = co.dtm_id AND co.row_number = 2
LEFT JOIN commercialoverlay AS co3
    ON ztl.dtm_id = co.dtm_id AND co.row_number = 3
```

I'd like to clean up this file a bit more, but think that'll take a bit of upstream logic as specified in #928

Reply:
And same for specialdist and zoningdist